scripts/collect_steps.py

```diff
@@ -0,0 +1,165 @@
+import os
+import re
+import subprocess
+
+# This script collects all the steps that are executed in the oneshot_ipa with
+# all possible configurations.
+
+IPA_ENV = [["RUST_LOG", "ipa=DEBUG"]]
+ARGS = [
+    "cargo",
+    "bench",
+    "--bench",
+    "oneshot_ipa",
+    "--no-default-features",
+    "--features=enable-benches debug-trace step-trace",
+    "--",
+    "--num-multi-bits",
+    "1",
+]
+QUERY_SIZE = 10
+PER_USER_CAP = [1, 3]
+ATTRIBUTION_WINDOW = [0, 86400]
+BREAKDOWN_KEYS = [1, 33]
+SECURITY_MODEL = ["malicious", "semi-honest"]
+ROOT_STEP_PREFIX = "protocol/alloc::string::String::run-0"
```

Comment:
This is what IPA currently outputs as the root step. I don't have enough context but happy to change it if need to .

Reply:
that is really weird, this is what I am seeing

```bash
Using random seed: 11119392551709147875 for 10 records
Preparation complete in 4.033209ms
Malicious IPA for 10 records took 416.62925ms
Step,Records sent,Bytes sent,Indexed PRSS,Sequential PRSS
protocol/run-0,0,0,9,0
protocol/run-0/after_convert_all_bits,0,0,9,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_credit,27,108,27,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_credit/duplicate_multiply,27,108,27,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_credit/randomness_for_validation,0,0,27,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_stop_bit,24,96,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_stop_bit/duplicate_multiply,24,96,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth0/current_stop_bit_times_successor_stop_bit/randomness_for_validation,0,0,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth1/current_stop_bit_times_successor_credit,24,96,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth1/current_stop_bit_times_successor_credit/duplicate_multiply,24,96,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth1/current_stop_bit_times_successor_credit/randomness_for_validation,0,0,24,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth1/current_stop_bit_times_successor_stop_bit,18,72,18,0
protocol/run-0/after_convert_all_bits/malicious_protocol/accumulate_credit/depth1/current_stop_bit_times_successor_stop_bit/duplicate_multiply,18,72,18,0
```

it may be a degradation from the step changes

Reply:
The extra "alloc::..." part is added when `step-trace` feature is enabled which is added in #655 .